src/hooks/useComments.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+
+import { Comment } from '../types';
+import * as commentsService from '../api/comments';
+
+export const useComments = (selectedPostId: number) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadComments = () => {
+    setErrorMessage('');
+    setIsLoading(true);
+
+    commentsService
+      .getComments(selectedPostId)
+      .then(setComments)
+      .catch(() => setErrorMessage('Something went wrong'))
```

Comment:
Consider providing more detailed error messages or logging the error to help with debugging. The current message 'Something went wrong' is generic and may not provide enough information for troubleshooting.